pandas/core/generic.py

```diff
@@ -4880,20 +4880,39 @@ def describe(self, percentiles=None, include=None, exclude=None):
 
         # median should always be included
         if (percentiles != 0.5).all():  # median isn't included
-            lh = percentiles[percentiles < .5]
-            uh = percentiles[percentiles > .5]
-            percentiles = np.hstack([lh, 0.5, uh])
-
-        def pretty_name(x):
-            x *= 100
-            if x == int(x):
-                return '%.0f%%' % x
-            else:
-                return '%.1f%%' % x
+            percentiles = np.concatenate((percentiles, [0.5]))
```

Comment:
It's not visible very well but there is an indent. So it's called if `(percentiles != 0.5).all()`. I postpone the unique check until later to sort only once and skip inserting 0.5 in the middle of an array.